src/components/AvatarMenu/AvatarMenu.test.tsx

```diff
@@ -0,0 +1,97 @@
+import { render, screen, waitFor } from '@testing-library/react';
+import '@testing-library/jest-dom';
+import { AvatarMenu } from './AvatarMenu';
+import type { AvatarMenuItem } from './AvatarMenu.types';
+import { beforeEach, describe, expect, test, vi } from 'vitest';
+import userEvent from '@testing-library/user-event';
+
+const mockLogout = vi.fn();
+const mockSettings = vi.fn();
+
+const mockMenuItems: AvatarMenuItem[] = [
+  { label: 'Configuración', onClick: mockSettings, 'data-testid': 'settings-option' },
+  { label: 'Cerrar Sesión', onClick: mockLogout, 'data-testid': 'logout-option', divider: true },
+];
+
+const mockProps = {
+  avatarUrl: 'https://test.com/avatar.jpg',
+  username: 'Test User',
+  menuItems: mockMenuItems,
+  'data-testid': 'user-avatar-button',
+};
+
+describe('AvatarMenu', () => {
+  beforeEach(() => {
+    vi.clearAllMocks();
+  });
+  const getAvatarButton = (username: string = mockProps.username) => {
+    return screen.getByRole('button', {
+      name: `Menú de usuario${username ? `: ${username}` : ''}`
+    });
```

Comment:
Tests use a Spanish aria-label check to find the button (getByRole('button', { name: `Menú de usuario...` })). The project mandates English as the codebase standard; tests and component labels should be in English. Also relying on exact aria-label strings in multiple tests couples tests tightly to text that may change (i18n).

**Sugerencia:**
```typescript
Change component aria-label to English (e.g. 'User menu') and update tests accordingly. Prefer using aria-label constants or data-testid for stable targeting, and prefer role-based selectors like getByRole('button', { name: /user menu/i }) for resilience.
```